app/pages/products/sms/sms.service.ts

```diff
@@ -0,0 +1,80 @@
+declare var require;
+import { OvhRequestService } from '../../../services/ovh-request/ovh-request.service';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/toPromise';
+import 'rxjs/add/operator/zip';
+
+const moment = require('moment');
+
+@Injectable()
+export class SmsService {
+  constructor(private ovhRequest: OvhRequestService) {
+
+  }
+
+  get() {
+    return this.ovhRequest.get('/sms');
+  }
+
+  getInfos(serviceName: string) {
+    return this.ovhRequest.get(`/sms/${serviceName}`);
+  }
+
+  putInfos(serviceName: string, data: any) {
+    return this.ovhRequest.put(`/sms/${serviceName}`, JSON.stringify(data));
+  }
+
+  getServiceInfos(serviceName: string) {
+    return this.ovhRequest.get(`/sms/${serviceName}/serviceInfos`);
+  }
+
+  getTasks(serviceName: string) {
+    return this.ovhRequest.get(`/sms/${serviceName}/task`);
+  }
+
+  getTask(serviceName: string, id: number) {
+    return this.ovhRequest.get(`/sms/${serviceName}/task/${id}`);
+  }
+
+  getIncomings(serviceName: string) {
+    return this.ovhRequest.get(`/sms/${serviceName}/incoming`, {
+      search: {
+        'creationDatetime.from': moment().startOf('month').format('YYYY-MM-DD'),
+        'creationDatetime.to': moment().endOf('month').format('YYYY-MM-DD')
+      }
+    }).toPromise()
```

Comment:
It's done. Let me know if I misunderstood something about *observable* and so on ><.